### DISCUSSION TITLE
ssr not effective

### DISCUSSION THREAD

**Discussion by zhaoyiming0803** [gatsbyjs/gatsby#34239] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I created a new project with command `gatsby new`，edit following `src/pages/index.js`，run `npm run develop`，but i am not see the source code in browser `view-source:http://localhost:8000/`, has not content `SSR Page with Dogs`
``` javascript
import * as React from "react"

export async function getServerData() {
  try {
    const res = await fetch(`https://dog.ceo/api/breeds/image/random`)
    if (!res.ok) {
      throw new Error(`Response failed`)
    }
    return {
      props: await res.json(),
    }
  } catch (error) {
    return {
      status: 500,
      headers: {},
      props: {}
    }
  }
}

const SSRPage = ({ serverData }) => (
  <main>
    <h1>SSR Page with Dogs</h1>
    <img alt="Happy dog" src={serverData.message} />
  </main>
)

export default SSRPage

```

### Reproduction Link

https://github.com/zhaoyiming0803/test-gatsbyjs

### Steps to Reproduce

1、`gatsby new`
2、edit following `src/pages/index.js`
3、run `npm run develop` 
4、not see the source code in browser `view-source:http://localhost:8000/`, has not content `SSR Page with Dogs`

### Expected Result

see the source code in browser `view-source:http://localhost:8000/`, has not content `SSR Page with Dogs`

### Actual Result

use ssr，but not see the source code in browser `view-source:http://localhost:8000/`, has not content `SSR Page with Dogs`

### Environment

```shell
json
{
"node": "v14.18.2"
"gatsby": "^4.3.0",
"react": "^17.0.1",
 "react-dom": "^17.0.1"
}
```
```


### Config Flags

``` javascript
module.exports = {
  siteMetadata: {
    siteUrl: "https://www.yourdomain.tld",
    title: "My Gatsby Site",
  },
  plugins: []
};

```

**Comment by LekoArts** [comment, marked as answer]
Hi!

`gatsby develop` is a development environment so it only renders things on the client. The thing you want to test out can be done with `gatsby build` & `gatsby serve`.

**Reply by zhaoyiming0803** [reply]
thanks!

**Comment by rwitchell** [comment]
to provide another answer for this: turn off javascript for your tab and then reload the page. You should then see the "SSR Page with Dogs" content load (and nothing else)
